localstack/services/edge.py

```diff
@@ -147,6 +150,53 @@ def do_forward_request_network(port, method, path, data, headers):
     return response
 
 
+def get_auth_string(headers, data=None):
+    """
+    Get Auth header from Header (this is how aws client's like boto typically
+    provide it) or from query string or url encoded parameters (sometimes
+    happens with presigned requests. Always return in the Authorization Header
+    form.
+
+    Typically an auth string comes in as a header:
+
+        Authorization: AWS4-HMAC-SHA256 \
+        Credential=_not_needed_locally_/20210312/us-east-1/sqs/aws4_request, \
+        SignedHeaders=content-type;host;x-amz-date, \
+        Signature=9277c941f4ecafcc0f290728e50cd7a3fa0e41763fbd2373fcdd3faf2dbddc2e
+
+    Here's what Authorization looks like as part of an presigned GET request:
+
+       &X-Amz-Algorithm=AWS4-HMAC-SHA256\
+       &X-Amz-Credential=test%2F20210313%2Fus-east-1%2Fsqs%2Faws4_request\
+       &X-Amz-Date=20210313T011059Z&X-Amz-Expires=86400000&X-Amz-SignedHeaders=host\
+       &X-Amz-Signature=2c652c7bc9a3b75579db3d987d1e6dd056f0ac776c1e1d4ec91e2ce84e5ad3ae
+
+    """
+
+    auth_header = headers.get('authorization', '')
+
+    if auth_header != '':
```

Comment:
nit: Perhaps a bit more pythonic - we could check for truthy/falsey values here, instead of comparing to empty values:
```
    if auth_header:
        return auth_header
    if not data:
        return ''
```